local-cli/generator/index.js

```diff
@@ -112,11 +119,31 @@ module.exports = yeoman.generators.NamedBase.extend({
       return;
     }
 
-    this.npmInstall(`react@${reactVersion}`, { '--save': true, '--save-exact': true });
+    isYarnInstalled(function(hasYarn) {
+      if (hasYarn) {
+        exec(`yarn add react@${reactVersion} --exact`, function(err, stdout, stderr) {
+          if (err) {
+            throw err;
+          }
+        });
+      } else {
+        this.npmInstall(`react@${reactVersion}`, { '--save': true, '--save-exact': true });
+      }
+    });
     if (!this.options['skip-jest']) {
-      this.npmInstall(`jest babel-jest jest-react-native babel-preset-react-native react-test-renderer@${reactVersion}`.split(' '), {
-        saveDev: true,
-        '--save-exact': true
+      isYarnInstalled(function(hasYarn) {
```

Comment:
I think we should run isYarnInstalled once and with execSync in the beginning of the script, it will be easier then